### PR TITLE
[sending] Change CheckedContinuation.resume and Async{Throwing,}Stream.yield to use transferring parameters.

### PR DESCRIPTION
**Pull request by gottesmm**
[transferring] Change CheckedContinuation.resume and Async{Throwing,}Stream.yield to use transferring parameters.

Importantly, I used silgen_name to preserve the name since I have not yet
removed the mangling for transferring for named entities. This ensures that
this is an ABI neutral change until I make that other change.

rdar://120420024

### CODE REVIEW

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
Found the type reconstruction issue. I need to do some more work here since we need to mark withCheckedContinuation as returning a transferring result.

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
But before I do that, I would like to see what the bots think about this without that part of the change.

File: test/DebugInfo/transferring.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// RUN: %target-swift-frontend -emit-ir -g -o /dev/null -module-name test -strict-concurrency=complete -swift-version 5 -enable-experimental-feature TransferringArgsAndResults -disable-availability-checking %s
+
```
Comment (adrian-prantl) [inline test/DebugInfo/transferring.swift:2]: can you add a CHECK: that ensures that the mangled name you're looking for appears in a DI*Type?

**Comment by gottesmm** [comment]
Just doing a test. I still have to make sending parsed without the flag enabled, but the rest is here.

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
Need to fix one thing for Adrian. Going to push and restart testing... but on macOS/Linux we got through the Swift tests

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test linux platform

**Comment by gottesmm** [comment]
There might have been a mid-air collision with sourcekit-lsp

**Comment by gottesmm** [comment]
@swift-ci smoke test linux platform

**Pull request merged**